src/main/java/tetris/AbstractShape.java

```diff
@@ -2,16 +2,14 @@
 
 public abstract class AbstractShape {
 
-    protected int[] x, y;
+    private Coordinate[] minos;
```

Comment:
Maybe create something in between Tetromino and Coordinate, namely Mino. Then a Mino can contain its own color and it makes collision handling easier if we want to make more advanced collision rules such as the posibility to automatically shift 1 pos left/right during a rotation. This also allows to make the grid a Mino[][] grid, such that we can later add more properties to the Mino like age for example.